docs/specs/NETWORKING.md

```diff
@@ -1,3 +1,769 @@
+  "error": "EntryNotFoundError"
+}
+```
+
+The following error codes exist: **TBD**
+- `EntryNotFoundError` is returned when no results were found
+- `InternalError` is returned upon internal error
+
+### Protocols
+
+SSV nodes use the following stream protocols:
+
+### 1. Highest Decided
+
+This protocol is used by a node to find out what is the highest decided message for a specific validator.
+In case there are no decided messages, it will return an empty array of messages.
+
+`/ssv/sync/decided/highest/0.0.1`
+
+<details>
+  <summary>examples</summary>
+
+  Request:
+  ```json
+  {
+    "identifier": "...",
+    "type":   0,
+  }
+  ```
+
+  Response:
+  ```json
+  {
+    "messages": [
+      {
+        "message": {
+          "type": 3,
+          "round": 1,
+          "identifier": "...",
+          "seq_number": 7943,
+          "value": "Xmcg...sPM="
+        },
+        "signature": "g5y....7Dv",
+        "signer_ids": [1,2,4]
+      }
+    ],
+    "type": 0,
+    "identifier": "..."
+  }
+  ```
+</details>
+
+### 2. Decided By Range
+
+This protocol enables to sync decided messages in some specific range.
+
+The request should specify the desired range, while the response will include all the found messages for that range.
+
+`/ssv/sync/decided/range/0.0.1`
+
+<details>
+  <summary>examples</summary>
+  
+  Request:
+  ```json
+  {
+    "identifier": "...",
+    "params": [1200, 1225],
+    "type":   1,
+  }
+  ```
+
+  Response:
+  ```json
+  {
+    "identifier": "...",
+    "params": [1200, 1225],
+    "messages": [
+      {
+        "message": {
+          "type": 3,
+          "round": 1,
+          "identifier": "...",
+          "seq_number": 1200,
+          "value": "Xmcg...sPM="
+        },
+        "signature": "g5y....7Dv",
+        "signer_ids": [1,2,4]
+      },
+      // ... 1201-1224
+      {
+        "message": {
+          "type": 3,
+          "round": 1,
+          "identifier": "...",
+          "seq_number": 1225,
+          "value": "Xmcg...sPM="
+        },
+        "signature": "g5y....7Dv",
+        "signer_ids": [1,2,4]
+      }
+    ],
+    "type":   1,
+  }
+  ```
+</details>
+
+
+### 3. Last Change Round
+
+This protocol enables a node to catch up with change round messages.
+
+`/ssv/sync/last_change_round/0.0.1`
+
+<details>
+  <summary>examples</summary>
+  
+  Request:
+  ```json
+  {
+    "identifier": "...",
+    "params": [7554],
+    "type":   2,
+  }
+  ```
+
+  Response:
+  ```json
+  {
+    "identifier": "...",
+    "params": [7554],
+    "messages": [
+      {
+        "message": {
+          "type": 4,
+          "round": 6,
+          "identifier": "...",
+          "seq_number": 7554,
+          "value": "Xmcg...sPM="
+        },
+        "signature": "g5y....7Dv",
+        "signer_ids": [1]
+      }
+    ],
+    "type":   2,
+  }
+  ```
+</details>
+
+---
+
+## Handshake protocol
+
+`/ssv/handshake/0.0.1`
+
+The handshake protocol allows peers to identify, by exchanging signed information. \
```

Comment:
I'm not sure I get the handshake messages. The way I see ethereum doing it is with status, goodbye and ping messages. Is that what we are trying to get here?